opensrp-reveal/src/main/java/org/smartregister/reveal/model/TaskDetails.java

```diff
@@ -102,6 +117,50 @@ public Integer getTaskCount() {
         return taskCount;
     }
 
+    public boolean isFamilyRegistered() {
+        return familyRegistered;
+    }
+
+    public void setFamilyRegistered(boolean familyRegistered) {
+        this.familyRegistered = familyRegistered;
+    }
+
+    public boolean isBednetDistributed() {
+        return bednetDistributed;
+    }
+
+    public void setBednetDistributed(boolean bednetDistributed) {
+        this.bednetDistributed = bednetDistributed;
+    }
+
+    public boolean isBloodScreeningDone() {
+        return bloodScreeningDone;
+    }
+
+    public void setBloodScreeningDone(boolean bloodScreeningDone) {
+        this.bloodScreeningDone = bloodScreeningDone;
+    }
+
+    public void setGroupedTaskCodeStatus(String groupedTaskCodeStatusString) {
+        setFamilyRegistered(false);
+        setBednetDistributed(false);
+        setBloodScreeningDone(false);
+        if (StringUtils.isEmpty(groupedTaskCodeStatusString)) {
+            return;
+        }
+        String[] groupedTaskCodeStatusArray = groupedTaskCodeStatusString.split(COMMA);
+        for (int i = 0; i < groupedTaskCodeStatusArray.length; i++) {
+            String[] taskCodeStatusArray = groupedTaskCodeStatusArray[i].split(HYPHEN);
+            if (taskCodeStatusArray[0].equals(REGISTER_FAMILY) && taskCodeStatusArray[1].equals(COMPLETE)) {
```

Comment:
Could you check that the split array if of size two.
Also put the constants as first part of the equals for null safety e.g 
`if (REGISTER_FAMILY.equals(taskCodeStatusArray[0]) && COMPLETE.equals(taskCodeStatusArray[1])) `